### DISCUSSION TITLE
Add documentation about docker images

### DISCUSSION THREAD

**Discussion by evroon** [evroon/bracket#445] [Q&A]

**Comment by officialdanielamani** [comment]
About Docker Image, Is there any example of docker compose as I using Portainer. I try using dockerfile to compile image but there seams error in Pipeline

**Comment by evroon** [comment]
@officialdanielamani Thanks for your interest, in #208 I am trying to get the docker compose config to work, but it's not finished yet.

I can try to get it to work in a few weeks, but don't have time right now.

**Comment by evroon** [comment]
@officialdanielamani the docker-compose config should now work.
See: https://github.com/evroon/bracket/blob/master/docker-compose.yml

You should be able to start Bracket using the commands written here:
https://github.com/evroon/bracket#quickstart

**Comment by officialdanielamani** [comment]
Just asking, I want to try the application, the docker is up and running but I greeted by login screen. How I want to configure admin or  login the application as signup is disable in this stage?

I try add ci.env on bracket-backend but get some error. Is I am doing wrong?

![Screenshot 2023-09-24 194420](https://github.com/evroon/bracket/assets/102462043/94844853-59be-4b40-a397-916d54284c5b)

**Comment by evroon** [comment]
Ah I realized I didn't write what credentials to use, I have specified that now in the README (https://github.com/evroon/bracket/pull/283). 
The error you see is indeed a bug, thanks for notifying me. I have merged a fix here: https://github.com/evroon/bracket/pull/282

I created a [new release](https://github.com/evroon/bracket/releases/tag/v0.1.9), so if you pull the new docker images, it should work. (`sudo docker-compose pull`)

It shouldn't be necessary to add `ci.env` or such files, you can specify the environment variables in the `docker-compose.yml`. So if you want to change the admin password, add `ADMIN_PASSWORD: "my password"` here:

https://github.com/evroon/bracket/blob/master/docker-compose.yml#L18

But then you need to recreate the database, by deleting and recreating the database container for example.

**Comment by officialdanielamani** [comment]
Thanks for update, but I encounter a problem. Example I run the docker on a PC-(A) than access the docker from another PC-(B), I cannot login, it give `Failed to load resource: net::ERR_BLOCKED_BY_CLIENT localhost:8400/token:1` on Inspect element console. Do I need to configure CORS_ORIGINS or this version only run on localhost?

![image](https://github.com/evroon/bracket/assets/102462043/5dc5df7b-6e26-4809-b4e6-21b8782b58ec)

**Comment by evroon** [comment]
I think `CORS_ORIGINS` should just contain the IP address of the PC that hosts the webserver (so PC A).
But I haven't tested it with IP addresses on a local network, only domain names on the public internet, for which it works.

**Comment by GithubPatrick91** [comment]
I could used it from remote by adding these environment variables `CORS_ORIGINS` and `NEXT_PUBLIC_API_BASE_URL` to the docker-compose file:

```
bracket-frontend:
        ...
        environment:
            NEXT_PUBLIC_API_BASE_URL: 'http://HOST_PC_IP:8400'
       ...

    bracket-backend:
        ...
        environment:
            ENVIRONMENT: "DEVELOPMENT"
            CORS_ORIGINS: '*'
            PG_DSN: "postgresql://bracket_dev:bracket_dev@postgres:5432/bracket_dev"
        ...
```

**Comment by evroon** [comment]
I have updated docker-compose.yml and also the docs: https://evroon.github.io/bracket/docs/running-bracket/deployment/

I have disabled the CORS middleware for the DEV environment. Also I added `NEXT_PUBLIC_API_BASE_URL` to the docs and example docker-compose.yml to explain how to run the frontend on somewhere else than localhost

Does this solve the unclarity? For now, I will turn this into a discussion because it's more about help than a specific feature/bug.
